adoc/admin-security-certificates.adoc

```diff
@@ -470,6 +483,272 @@ its `pki` folder. This file is a symlink to the latest generated certificate.
 Restarting the kubelet effectively rotates the certificate to read the latest
 generated file.
 
+[[addon_certificates_rotation]]
+=== Addon certificates rotation
+
+The addon certificate could be automatically rotated by leverages the opensource solution `cert-manager` and `reloader`. The cert-manager is to auto rotates the addon certificates stored in Secret, and the reloader is to watch the addon certificate Secret and rolling upgrade the addon Deployments or DaemonSet.
+
+* Prerequisites
+
+. To let reloader auto rolling upgrade the addon Deployments or DaemonSet, we need to label the addons:
++
+[source,bash]
+----
+kubectl annotate --overwrite deployment/oidc-dex -n kube-system secret.reloader.stakater.com/reload=oidc-dex-cert
+
+kubectl annotate --overwrite deployment/oidc-gangway -n kube-system secret.reloader.stakater.com/reload=oidc-gangway-cert
+
+kubectl annotate --overwrite deployment/metrics-server -n kube-system secret.reloader.stakater.com/reload=metrics-server-cert
+----
+
+. Upload the Kubernetes CA cert/key pair to Secret in namespace kube-system:
++
+[source,bash]
+----
+kubectl create secret tls kubernetes-ca --cert=pki/ca.crt --key=pki/ca.key -n kube-system
+----
++
+[NOTE]
+====
+If you want to use custom trusted CA certificate/key to sign the certificate, upload to the secret resource.
+
+[source,bash]
+----
+kubectl create secret tls custom-trusted-ca --cert=<CUSTOM_TRUSTED_CA_CERT> --key=<CUSTOM_TRUSTED_CA_KEY> -n kube-system
+----
+====
+
+. Install reloader by helm chart:
++
+[source,bash]
+----
+helm install suse/reloader \
+    --name ${RELEASE_NAME} \
+    --namespace cert-manager
+----
+
+. Install cert-manager by helm chart:
++
+[source,bash]
+----
+helm install jetstack/cert-manager \
+    --name ${RELEASE_NAME} \
+    --namespace cert-manager \
+    --set global.leaderElection.namespace=cert-manager \
+    --set installCRDs=true
+----
+
+* Cert-Manager CA Issuer Resource
++
+The cert-manager CA issuer is a Kubernetes resource that represent certificate authority (CA) that are able to generate signed certificates by honoring certificate signing requests (CSR). The cert-manager certificates require a referenced issuer that is in a ready condition to attempt to honor the request.
++
+[NOTE]
+====
+An `Issuer` is a namespaced resource, and it is not possible to issue certificates from an Issuer in a different namespace. This means you will need to create an Issuer in each namespace you wish to obtain Certificates in.
+
+If you want to create a single Issuer that can be consumed in multiple namespaces, you should consider creating a `ClusterIssuer` resource. This is almost identical to the Issuer resource, however is non-namespaced so it can be used to issue Certificates across all namespaces.
```

Comment:
```suggestion
If you want to create a single Issuer that can be consumed in multiple namespaces, you should consider creating a `ClusterIssuer` resource. This is almost identical to the Issuer resource, however, it is cluster-wide so it can be used to issue certificates in all namespaces.
```